openhands/sdk/agent/agent.py

```diff
@@ -285,14 +285,43 @@ def step(
                 self._execute_actions(state, action_events, on_event)
 
         else:
-            logger.info("LLM produced a message response - awaits user input")
+            logger.info("LLM produced a message response - converting to FinishAction")
             state.agent_status = AgentExecutionStatus.FINISHED
-            msg_event = MessageEvent(
+
+            # Extract message content to use as finish message
+            message_text = ""
+            for content in message.content:
+                if isinstance(content, TextContent):
+                    message_text += content.text
+
+            # Create a FinishAction with the message content
+            finish_action = FinishAction(message=message_text or "Task completed")
+
+            # Create ActionEvent for the finish action
+            action_event = ActionEvent(
                 source="agent",
-                llm_message=message,
+                thought=cast(list[TextContent], message.content),
+                reasoning_content=message.reasoning_content,
+                action=finish_action,
+                tool_name="finish",
+                tool_call_id="auto_finish_" + response.id,
```

Comment:
@OpenHands we should NOT put tool call id here since this is manually simulated FinishAction. Is it easy to make sure ActionEvent is versatile enough for this behavior?